datafusion/physical-expr/src/aggregate/count_distinct.rs

```diff
@@ -336,9 +336,18 @@ where
     }
 
     fn size(&self) -> usize {
+        let estimated_buckets = (self.values.len().checked_mul(8).unwrap_or(usize::MAX)
```

Comment:
maybe we can (as a follow on PR) put this logic into its own function (with comments) as estimating the size of hashbrown hashtables is likely to come up again